Sources/KlaviyoForms/InAppForms/IAFPresentationManager.swift

```diff
@@ -118,26 +119,44 @@ class IAFPresentationManager {
         viewController?.modalPresentationStyle = .overCurrentContext
     }
 
-    // MARK: - Form Event Subscription
+    // MARK: - Form Lifecycle Listener Setup
 
-    private func listenForFormEvents() {
+    private func setupFormLifecycleListener() {
         guard let viewModel else { return }
 
+        if #available(iOS 14.0, *) {
+            Logger.webViewLogger.info("👂 Starting to listen for form lifecycle events (BEFORE handshake)")
+        }
+
+        // Start listening for form lifecycle events before handshake to avoid missing any events
+        formEventTask = Task { [weak self] in
+            guard let self else { return }
+            for await event in viewModel.formLifecycleStream {
+                self.handleFormEvent(event)
+            }
+        }
+
         Task { [weak self] in
             guard let self else { return }
+            if #available(iOS 14.0, *) {
+                Logger.webViewLogger.info("🤝 Starting handshake with KlaviyoJS")
+            }
             do {
                 try await viewModel.establishHandshake(timeout: NetworkSession.networkTimeout.seconds)
+                if #available(iOS 14.0, *) {
+                    Logger.webViewLogger.info("✅ Handshake completed successfully. Waiting for forms data to load before replaying events...")
+                }
+                
+                // Wait up to 5 seconds for forms data to load
+                try? await Task.sleep(nanoseconds: 5_000_000_000) // 5 seconds
+                
+                // Replay events (either already replayed by formsDataLoaded, or now as fallback)
+                await self.replayPendingEvents()
```

Comment:
This is a fallback for when JS doesn't emit `formsDataLoaded`. Anyone think of any risk with this and also what y'all think about the timeout duration?

Reply:
Hmm good point. I'll remove it. It'll simplify the code a bit.